avocado/core/status/repo.py

```diff
@@ -52,10 +52,14 @@ def _set_task_data(self, message):
             self._all_data[task_id] = []
         self._all_data[task_id].append(message)
 
-    def get_task_data(self, task_id):
+    def get_all_task_data(self, task_id):
         """Returns all data on a given task, by its ID."""
         return self._all_data.get(task_id)
 
+    def get_task_data(self, task_id, index):
+        """Returns data on a given task, by its ID."""
+        return self._all_data.get(task_id)[index]
```

Comment:
I find this index style strange for a method signature. what is representing this index? How complex it is to call by names/properties?